source/mir/random/ndvariable.d

```diff
@@ -279,6 +280,126 @@ nothrow @safe version(mir_random_test) unittest
     assert(fabs(x[0] + x[1] + x[2] - 1) < 1e-10);
 }
 
+/++
+Multinomial distribution.
++/
+struct MultinomialVariable(U:size_t, T)
+    if (isFloatingPoint!T)
+{
+    import mir.random.variable : binomialVar;
+
+    ///
+    enum isNdRandomVariable = true;
+    ///
+    alias Element = T;
+
+    ///
+    const(T)[] probs;
+    const size_t N;
+
+    /++
+    Params:
+        probs = probabilities
+        N = Number of rolls
+    Constraints: `sum(probs[i]) <= 1`
+    +/
+    this()(const size_t N, const(T)[] probs)
+    {
+        this.N = N;
+        this.probs = probs;
+    }
+
+    ///
+    pragma(inline, false)
+    size_t[] opCall(G)(scope ref G gen, scope U[] result)
+        if (isSaturatedRandomEngine!G)
+    {
+        uint k;
+        alias p =this.probs;
+        auto K = p.length;
+        double norm = 0.0;
+        double sum_p = 0.0;
+        alias rng = rne;
+
+        uint sum_n = 0;
+        size_t[] n;
```

Comment:
no need this, the `result` should be initialized instead, it should be already preallocated.

Reply:
done